docs/user-guide/concepts/agents/session-management.md

```diff
@@ -0,0 +1,265 @@
+    - `updated_at`: ISO format timestamp of when the message was last updated
+
+These data models work together to provide a complete representation of an agent's state and conversation history. The session management system handles serialization and deserialization of these models, including special handling for binary data using base64 encoding.
+
+### 3. Storage Structure
+
+When using `FileSessionManager`, sessions are stored in the following directory structure:
+
+```
+/<sessions_dir>/
+└── session_<session_id>/
+    ├── session.json                # Session metadata
+    └── agents/
+        └── agent_<agent_id>/
+            ├── agent.json          # Agent metadata and state
+            └── messages/
+                ├── message_<created_timestamp>_<id1>.json
+                └── message_<created_timestamp>_<id2>.json
+```
+
+When using `S3SessionManager`, a similar structure is maintained using S3 object keys.
+
+## Custom Session Repositories
+
+For advanced use cases, you can implement your own session storage backend by creating a custom session repository:
+
+```python
+from typing import Optional
+from strands import Agent
+from strands.session.repository_session_manager import RepositorySessionManager
+from strands.session.session_repository import SessionRepository
+from strands.types.session import Session, SessionAgent, SessionMessage
+
+class CustomSessionRepository(SessionRepository):
+    """Custom session repository implementation."""
+    
+    def __init__(self):
+        """Initialize with your custom storage backend."""
+        # Initialize your storage backend (e.g., database connection)
+        self.db = YourDatabaseClient()
+    
+    def create_session(self, session: Session) -> Session:
+        """Create a new session."""
+        self.db.sessions.insert(asdict(session))
+        return session
+    
+    def read_session(self, session_id: str) -> Optional[Session]:
+        """Read a session by ID."""
+        data = self.db.sessions.find_one({"session_id": session_id})
+        if data:
+            return Session.from_dict(data)
+        return None
+    
+    # Implement other required methods...
+    # create_agent, read_agent, update_agent
+    # create_message, read_message, update_message, list_messages
+
+# Use your custom repository with RepositorySessionManager
+custom_repo = CustomSessionRepository()
+session_manager = RepositorySessionManager(
+    session_id="user-789",
+    session_repository=custom_repo
+)
+
+agent = Agent(session_manager=session_manager)
+```
+
+This approach allows you to store session data in any backend system while leveraging the built-in session management logic.
+
+## Session Persistence Best Practices
+
+When implementing session persistence in your applications, consider these best practices:
+
+1. **Use Unique Session IDs**: Generate unique session IDs for each user or conversation context to prevent data overlap.
+
+2. **Consider Storage Requirements**:
```

Comment:
I might ditch 2 & 3 just so that we don't have a growing list of "storage providers"-specific tips.